### PR TITLE
feat: `exec` and optimze command processing logic

### PR DESCRIPTION
**Pull request by jinzhongjia**
- Move main's command logic into `command.zig` file
- `build.zig` adds run step, now you can directly use `zig build run -- ls` for testing
- `exec` experimental support

### CODE REVIEW

**Comment by jinzhongjia** [comment]
In fact, the current processing module is not elegant, I will try to make some more modifications

**Comment by jinzhongjia** [comment]
In addition, I recommend using rebase or squash. Merge causes the git roadmap to not look good.

**Comment by jinzhongjia** [comment]
image

About the exec feature, as mentioned earlier, https://github.com/hendriknielaender/zvm/issues/48

But the logic seems to need to be optimized

File: src/command.zig [hunk 1]
```diff
@@ -16,32 +18,100 @@ pub const Command = enum {
...
+    .{ .short_handle = "ls", .handle = "list", .cmd = Command.List },
+    .{ .short_handle = "i", .handle = "install", .cmd = Command.Install },
+    .{ .short_handle = null, .handle = "use", .cmd = Command.Use },
+    .{ .short_handle = null, .handle = "--version", .cmd = Command.Version },
+    .{ .short_handle = null, .handle = "--help", .cmd = Command.Help },
+    .{ .short_handle = null, .handle = "--default", .cmd = Command.Default },
+};
+
+/// parse command and handle commands
+pub fn handleCommand(params: []const []const u8) !void {
```
Comment (hendriknielaender) [inline src/command.zig:43]:
```suggestion
pub fn handle_command(params: []const []const u8) !void {
```

I did a previous PR, where i tried to follow tigerstyle. https://github.com/tigerbeetle/tigerbeetle/blob/main/docs/TIGER_STYLE.md#naming-things

I like the approach of keeping it simple and also use snake_case for fn.

Comment (hendriknielaender) [inline src/command.zig:43]: What do you think of write it like this?

```zig
pub fn handle_command(params: []const []const u8) !void {
    if (builtin.os.tag != .windows) {
        const command = std.fs.path.basename(params[0]);
        if (std.mem.eql(u8, command, "zig")) {
            try handle_alias(params);
        }
    }

    const commandData = get_command_data(params[1..]) catch |err| {
        // handle error appropriately
        return err;
    };

    switch (commandData.cmd) {
        .List => try handle_list(),
        .Install => try install_version(commandData.param),
        .Use => try use_version(commandData.param),
        .Default => try set_default(),
        .Version => try get_version(),
        .Help => try display_help(),
        .Unknown => try handle_unknown(),
    }
}

fn get_command_data(args: []const []const u8) !CommandData {
    for (args) |arg, index| {
        const opt = findOption(arg) orelse continue;

        return CommandData{
            .cmd = opt.cmd,
            .param = if (index + 1 < args.len) args[index + 1] else null,
        };
    }
    return CommandData{ .cmd = .Unknown };
}

fn find_option(arg: []const u8) ?OptionType {
    for (command_opts) |opt| {
        if (std.mem.eql(u8, arg, opt.handle)) return opt;
        if (opt.short_handle) |short_handle| {
            if (std.mem.eql(u8, arg, short_handle)) return opt;
        }
    }
    return null;
}
```

Comment (jinzhongjia) [reply src/command.zig:43]: The logic of this code is not complicated, I don't think it needs to be disassembled

Comment (jinzhongjia) [reply src/command.zig]: This is also the reason why I merged the original functions into one

**Pull request merged**